src/wallet/rpcwallet.cpp

```diff
@@ -3977,7 +3978,7 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
                 maxedOutFlag = true;
             } else {
                 estimatedTxSize += increase;
-                ShieldCoinbaseUTXO utxo = {out.tx->GetHash(), out.i, nValue};
+                ShieldCoinbaseUTXO utxo = {out.tx->GetHash(), out.i, scriptPubKey, nValue};
```

Comment:
Why this change?

Reply:
`z_sendmany` etc. were written to use `signrawtransaction` (by marshalling to JSON, calling the RPC, and then unmarshalling), which looks up the `scriptPubKey` for each input. `TransactionBuilder` implements transparent input signing directly, so it needs to be given this value. When we refactor to address the `TODO` about using `CWallet::CommitTransaction`, we will end up needing `scriptPubKey` for the non-builder case as well.